lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1121,6 +1125,26 @@ sub _create_bam_flagstat {
     return 1;
 }
 
+sub create_bam_header {
+    my $self = shift;
+
+    if (-s $self->bam_header_path) {
+        return 1;
+    }
+
+    my $lock = $self->get_bam_lock;
```

Comment:
Should this check again for the `bam_header_path` to already exist after locking?